documentation/modules/cruise-control/proc-generating-rebalance-proposals.adoc

```diff
@@ -0,0 +1,119 @@
+
+You can then analyze the summary information in the optimization proposal and decide whether to implement it. 
+
+.Prerequisites
+
+* You have xref:proc-deploying-cruise-control-{context}[deployed Cruise Control] to your {ProductName} cluster.
+
+* You have configured xref:con-optimization-goals-{context}[optimization goals].
+
+.Procedure
+
+. Create a `KafkaRebalance` resource:
+
+.. To use the _default optimization goals_ defined in the `Kafka` resource, leave the `spec` property empty:
++
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-cluster-rebalance
+  labels: 
+    strimzi.io/cluster: my-cluster
+spec: {}
+----
+
+.. To set _user-provided optimization goals_ instead of using the default goals, edit the `spec.goals` property and enter one or more goals.
++
+In the following example, rack awareness and replica capacity are user-provided optimization goals:
++
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-cluster-rebalance
+  labels: 
+    strimzi.io/cluster: my-cluster
+spec:    
+    goals: >
+      RackAwareGoal,
+      ReplicaCapacityGoal
+----
+
+. Create or update the resource:
++
+[source,shell,subs="+quotes"]
+----
+oc apply -f _your-file_
+----
++
+The Cluster Operator requests the optimization proposal from Cruise Control.
+This might take a few minutes depending on the size of the Kafka cluster. 
+
+. Check the status of the `KafkaRebalance` resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl get kafka kafkarebalance -o jsonpath='{.status}'
+----
++
+Cruise Control returns one of two statuses:
+
+** `PendingProposal`: The rebalance operator is polling the Cruise Control API to check if the optimization proposal is ready.
+
+** `ProposalReady`: The optimization proposal is ready for you to review and implement, if desired.
+The status is followed by the JSON optimization proposal.
+
+. Review the JSON optimization proposal in the `status.summary` property of the `KafkaRebalance` resource.
++
+Here is an example proposal:
++
+[source,shell,subs="+quotes"]
+----
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaRebalance
+metadata:
+  name: my-cluster-rebalance
+  labels: 
+    strimzi.io/cluster: my-cluster
+spec:
+  # ...
+status:
+  conditions:
+  - lastTransitionTime: 2020-05-23T23:46:57+0000
+    status: "True"
+    type: Ready
+  observedGeneration: 4
+  summary: CHECK THE FORMATTING IS CORRECT
+  - numIntraBrokerReplicaMovements:0,
+  - excludedBrokersForLeadership:[],
+  - numReplicaMovements:31,
+  - onDemandBalancednessScoreAfter:84.08179174145127,
+  - onDemandBalancednessScoreBefore:79.74705957325753,
+  - intraBrokerDataToMoveMB:0,
+  - recentWindows:1,
+  - dataToMoveMB:0,
+  - monitoredPartitionsPercentage:100.0,
+  - excludedTopics:[],
+  - numLeaderMovements:12,
+  - excludedBrokersForReplicaMove:[]
+# ...
```

Comment:
@laidan6000 It might be worth you running this proceedure yourself on minikube/minishift etc. Just to get a feel for it and to add an example output which has changed slightly from above. For example the condition `type` and `status` fields have swapped. Also the docs refer to the OpProp summary as "JSON format" when the above is yaml fromat. I am not sure if K8s (or some client) is converting the JSON we give it to YAML so you should double check.

Reply:
OK

Reply:
I have not been able to verify that the output is correct.

Reply:
What do you mean? You couldn't get a rebalance working or that the output doesn't look like that?

Reply:
I didn't have time to test this out.

Reply:
Ok well this output is still wrong. I will have to run a quick test to make make sure this is the correct format.

Reply:
So if you use the `kubectl describe kafkarebalance <rebalance-name>` you get a status that looks like this:
```
Status:
  Conditions:
    Last Transition Time:  2020-05-19T13:50:12.533Z
    Status:                ProposalReady
    Type:                  State
  Observed Generation:     1
  Optimization Result:
    Data To Move MB:  0
    Excluded Brokers For Leadership:
    Excluded Brokers For Replica Move:
    Excluded Topics:
    Intra Broker Data To Move MB:         0
    Monitored Partitions Percentage:      100
    Num Intra Broker Replica Movements:   0
    Num Leader Movements:                 0
    Num Replica Movements:                26
    On Demand Balancedness Score After:   81.8666802863978
    On Demand Balancedness Score Before:  78.01176356230222
    Recent Windows:                       1
  Session Id:                             05539377-ca7b-45ef-b359-e13564f1458c
```

Reply:
Thanks - I've added the status above.

Both the steps now use the `describe` command:

`kubectl describe kafkarebalance _rebalance-name_`